runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -43,22 +44,34 @@
 
 /** SDK objects that will be represented at some later point within a {@link Components} object. */
 public class SdkComponents {
-  private final RunnerApi.Components.Builder componentsBuilder;
+  private final RunnerApi.Components.Builder componentsBuilder = RunnerApi.Components.newBuilder();
 
-  private final BiMap<AppliedPTransform<?, ?, ?>, String> transformIds;
-  private final BiMap<PCollection<?>, String> pCollectionIds;
-  private final BiMap<WindowingStrategy<?, ?>, String> windowingStrategyIds;
+  private final BiMap<AppliedPTransform<?, ?, ?>, String> transformIds = HashBiMap.create();
```

Comment:
why are these changes to initialize everything necessary, when they were not being used before? Is this a preferred style to initialized in the ctor?

Reply:
I have found both ways to initialize in Beam code. Did this change to avoid copy-paste initialization of these fields in new constructor.